docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,491 @@
+
+```bash
+## test existing token
+# Get Token path from AWS credentials mounted to pod
+TOKEN_PATH=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^web_identity_token_file" |\
+    awk '{print$3}')
+
+# Get Controler's pod
+CAPI_POD=$(oc get pods -n openshift-machine-api \
+    -l api=clusterapi \
+    -o jsonpath='{.items[*].metadata.name}')
+
+# Extract tokens from the pod
+TOKEN=$(oc exec -n openshift-machine-api ${CAPI_POD} \
+    -c machine-controller -- cat ${TOKEN_PATH})
+
+echo $TOKEN | awk -F. '{ print $2 }' | base64 -d 2>/dev/null | jq .iss
+
+IAM_ROLE=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^role_arn" |\
+    awk '{print$3}')
+
+echo $IAM_ROLE
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}"
+```
+
+### Backup existing state <a name="backup"></a>
+
+- Get Objects and existing authentication response
+
+```bash
+export BACKUP_PATH=$PWD/current-cluster
+mkdir $BACKUP_PATH
+
+oc get authentication -o yaml |tee -a $BACKUP_PATH/authentication.yaml
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}" \
+    | jq -r '.Credentials=""' \
+    | tee ${BACKUP_PATH}/identities.json
+```
+
+- Save the current IAM Roles
+
+```bash
+aws iam list-roles \
+    | jq -r  ".Roles[] | select(.RoleName | startswith(\"${CLUSTER_NAME}-openshift\"))" \
+    | tee ${BACKUP_PATH}/iam-roles.json
+```
+
+## Migrate
+
+### Setup new OIDC with CloudFront Distribution <a name="setup"></a>
+
+#### Create the CloudFront Distribution <a name="setup-cloudfront"></a>
+
+- Create the Origin Access Identity (OAI)
+
+```bash
+export DIR_CCO="./"
+export OIDC_BUCKET_PATH="/pvt"
+
+export OAI_CLOUDFRONT_ID=$(aws cloudfront create-cloud-front-origin-access-identity \
+    --cloud-front-origin-access-identity-config \
+    CallerReference="${OIDC_BUCKET_NAME}",Comment="OAI-${OIDC_BUCKET_NAME}" \
+    | jq -r .CloudFrontOriginAccessIdentity.Id)
+```
+
+- Create the CloudFront Distribution
+```bash
+# Should be updated to master before merging
+#wget https://raw.githubusercontent.com/openshift/cloud-credential-operator/master/docs/sts-oidc-cloudfront.json.tpl
+wget https://raw.githubusercontent.com/mtulio/cloud-credential-operator/doc-sts-updates/docs/sts-oidc-cloudfront.json.tpl
+
+cat sts-oidc-cloudfront.json.tpl \
+   | envsubst \
+   > ${DIR_CCO}/oidc-cloudfront.json
+
+
+export CLOUDFRONT_HOST=$(aws cloudfront create-distribution-with-tags \
+    --distribution-config-with-tags \
+    file://${DIR_CCO}/oidc-cloudfront.json \
+    | jq -r .Distribution.DomainName)
```

Comment:
Encountered issue when use jshu-sts instead of admin account.

[cloud-user@preserve-for-hive-test regular]$ export CLOUDFRONT_HOST=$(aws cloudfront create-distribution-with-tags \
>     --distribution-config-with-tags \
>     file://${DIR_CCO}/oidc-cloudfront.json \
>     | jq -r .Distribution.DomainName)

An error occurred (AccessDenied) when calling the CreateDistributionWithTags operation: User: arn:aws:iam::301721915996:user/jshu-sts is not authorized to perform: cloudfront:CreateDistribution on resource: arn:aws:cloudfront::301721915996:distribution/* because no identity-based policy allows the cloudfront:CreateDistribution action

Reply:
Done https://github.com/openshift/cloud-credential-operator/pull/515/commits/c8f24f602088c33f38f5cc1fd03430200eb0cc0b